src/main/java/com/klodnicki/Bike/v2/rest/controller/RentBikeController.java

```diff
@@ -0,0 +1,56 @@
+package com.klodnicki.Bike.v2.rest.controller;
+
+import com.klodnicki.Bike.v2.DTO.bike.BikeForNormalUserResponseDTO;
+import com.klodnicki.Bike.v2.DTO.bike.ListBikesForNormalUserResponseDTO;
+import com.klodnicki.Bike.v2.DTO.rent.RentRequestDTO;
+import com.klodnicki.Bike.v2.DTO.rent.RentResponseDTO;
+import com.klodnicki.Bike.v2.model.RentRequest;
+import com.klodnicki.Bike.v2.model.entity.Bike;
+import com.klodnicki.Bike.v2.model.entity.ChargingStation;
+import com.klodnicki.Bike.v2.service.interfacee.GenericRentBikeService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+@AllArgsConstructor
+public class RentBikeController {
+
+    private final GenericRentBikeService rentBikeService;
+
+
+    @GetMapping("/bikes")
+    public ListBikesForNormalUserResponseDTO findAvailableBikes() {
+        List<BikeForNormalUserResponseDTO> bikeDTOs = rentBikeService.findAvailableBikes();
+
+        return new ListBikesForNormalUserResponseDTO(bikeDTOs);
+    }
+
+    @GetMapping("/bikes/{id}")
+    public BikeForNormalUserResponseDTO findBikeForNormalUserById(@PathVariable Long id) {
+        return rentBikeService.findBikeForNormalUserById(id);
+    }
+
+    @PostMapping("/rentals/add")
+    public RentResponseDTO rentBike(@RequestBody RentRequest rentRequest) {
+        return rentBikeService.rent(rentRequest);
+    }
+
+    @PutMapping("/bikes/list/add/{id}")
+    public ChargingStation addBikeToList(@PathVariable("id") Long chargingStationId, @RequestBody Bike bike) {
+        return rentBikeService.addBikeToList(chargingStationId, bike);
+    }
```

Comment:
DTO should be used as return type of this controller endpoint.
DTO should be used to @RequestBody.
Other than that - is this endpoint really needed? What is the purpose of it?

Reply:
This method is actually deprecated. It was only for test purpose. I shall delete it right now! thanks!